ProcessQueue/__init__.py

```diff
@@ -306,38 +345,131 @@ def main(msg: func.QueueMessage):
 
                 if not frequency:
                     frequency = 0
-                    
+
                 if data_dictionary is not None:
-                    val_desc = next((row['value_description'] for row in data_dictionary if str(row['field_name']) == str(name) and str(row['code']) == str(value)), None)
+
+                    # Look up value description
+                    val_desc = next(
+                        (
+                            row["value_description"]
+                            for row in data_dictionary
+                            if str(row["field_name"]) == str(name)
+                            and str(row["code"]) == str(value)
+                        ),
+                        None,
+                    )
+
+                    # Grab data from the 'code' column in the data dictionary
+                    # 'code' can contain an ordinary value (e.g. Yes, No, Nurse, Doctor)
+                    # or it could contain one of our pre-defined vocab names
+                    # e.g. SNOMED, RxNorm, ICD9 etc.
+                    code = next(
+                        (
+                            row["code"]
+                            for row in data_dictionary
+                            if str(row["field_name"]) == str(name)
+                        ),
+                        None,
+                    )
+
+                    # If 'code' is in our vocab list, try and convert the ScanReportValue (concept code) to conceptID
+                    # If there's a faulty concept code for the vocab, fail gracefully and set concept_id to default (-1)
+                    if code in vocabs:

```

Comment:
Perhaps we should also add `and val_desc is None` to gracefully ignore anything where a valid code is supplied but the `value` has also been filled in - which indicates that the data dictionary is malformed or perhaps we have a code which matches the vocab list but shouldn't be mapped. Thoughts?

Reply:
Interesting idea.

I think maybe it would depend how the data dictionaries are used 'in the wild'. According to the current standards, there should only be 1 row in the data dictionary for a vocab-coded field; all values within that field belong to the vocabulary defined in the dictionary. If a data partner happened to supply what would technically be a value description but in reality be a field description, it would just get ignored.

On the other hand, a data provider could supply 1 row per value in the dictionary, even if it's coded under some kind of vocabulary. In this case I guess the question would be - would we want to bring across that value description into CCOM? In other words, would a value description be useful information to have (for human readability) even though the concept code (and therefore the conceptID is known?